avocado/utils/kernel.py

```diff
@@ -34,7 +35,7 @@ class KernelBuild(object):
     SOURCE = 'linux-{version}.tar.gz'
 
     def __init__(self, version, config_path=None, work_dir=None,
-                 data_dirs=None):
+                 data_dirs=None, src_url, do_install=None):
```

Comment:
IMO there's no need to have this as a class parameter.  Users are free to either call (or not) the `install()` method if they choose to have the kernel installed.

Reply:
Should we turn this as method parameter instead ?

Reply:
I'd move `src_url` to `download()`, and make it optional, something like:

```diff
diff --git a/avocado/utils/kernel.py b/avocado/utils/kernel.py
index 654530ee..30a11b00 100644
--- a/avocado/utils/kernel.py
+++ b/avocado/utils/kernel.py
@@ -62,13 +62,20 @@ class KernelBuild(object):
                                               self.config_path,
                                               self.work_dir)
 
-    def download(self):
+    def download(self, url=None):
         """
         Download kernel source.
+
+        :param url: override the url from where to fetch the kernel
+                    source tarball
+        :type url: str or None
         """
-        self.kernel_file = self.SOURCE.format(version=self.version)
-        base_url = self.URL.format(major=self.version.split('.', 1)[0])
-        full_url = base_url + self.kernel_file
+        kernel_file = self.SOURCE.format(version=self.version)
+        if url is not None:
+            base_url = self.URL.format(major=self.version.split('.', 1)[0])
+        else:
+            base_url = url
+        full_url = base_url + kernel_file
         self.asset_path = asset.Asset(full_url, asset_hash=None,
                                       algorithm=None, locations=None,
                                       cache_dirs=self.data_dirs).fetch()
```

That way, it won't be required, but still can be overridden.